### PR TITLE
Prepare Client/Server implementation

### PR DESCRIPTION
**Pull request by lechld**

### CODE REVIEW

File: app/src/main/java/at/aau/edu/wizards/MainActivity.kt [hunk 1]
```diff
@@ -15,11 +18,26 @@ class MainActivity : AppCompatActivity() {
...
+
+    private fun handlePermissions() {
+        val permissionHandler = PermissionHandler(
+            activity = this,
+            permissions = REQUIRED_PERMISSIONS
+        ) { permissions ->
+            val anyDenied = permissions.any { !it.value }
+
+            println("Any permissions denied = $anyDenied")
+            // TODO: Need to properly handle denied permissions. App won't work without
```
Comment (Kirschzelle) [inline app/src/main/java/at/aau/edu/wizards/MainActivity.kt:39]: Same question about what to do with todos. (*badumz*) Create backlog tasks?

**Review by Kirschzelle — APPROVED** [review]
Looks good, approved. There seems to be some weird coverage in some files (Ex.: ClientImpl.kt), but that is negligible. Also see comment and feel free to merge.

File: app/src/main/java/at/aau/edu/wizards/MainActivity.kt (hunk 1 above)
Comment (lechld) [reply app/src/main/java/at/aau/edu/wizards/MainActivity.kt:39]: yadayada

Comment (Kirschzelle) [reply app/src/main/java/at/aau/edu/wizards/MainActivity.kt:39]: fairs

**Pull request merged**